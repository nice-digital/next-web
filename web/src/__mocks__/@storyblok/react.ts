export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
