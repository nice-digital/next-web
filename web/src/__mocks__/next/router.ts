export const useRouter = jest.fn().mockImplementation(() => {
	throw new Error("Provide a mock of useRouter for testing purposes");
});
