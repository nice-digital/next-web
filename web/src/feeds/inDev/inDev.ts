import { serverRuntimeConfig } from "@/config";

import { getFeedBodyCached, getFeedBodyUnCached } from "..";
import { ErrorResponse } from "../publications/types";

import {
	Project,
	FeedPath,
	AllProjects,
	Consultation,
	InConsultationProjects,
	ProjectDetail,
	IndevPanel,
} from "./types";

export * from "./types";

const cacheKeyPrefix = "inDev",
	{ defaultTTL, longTTL } = serverRuntimeConfig.cache,
	{ origin, apiKey } = serverRuntimeConfig.feeds.inDev;

export const getAllProjects = async (): Promise<Project[]> =>
	await getFeedBodyCached<Project[]>(
		cacheKeyPrefix,
		FeedPath.AllProjects,
		defaultTTL,
		async () =>
			(
				await getFeedBodyUnCached<AllProjects>(
					origin,
					FeedPath.AllProjects,
					apiKey
				)
			).embedded.niceIndevIndevelopmentProject
	);

export const getAllConsultations = async (): Promise<Consultation[]> =>
	await getFeedBodyCached<Consultation[]>(
		cacheKeyPrefix,
		FeedPath.InConsultationProjects,
		defaultTTL,
		async () =>
			(
				await getFeedBodyUnCached<InConsultationProjects>(
					origin,
					FeedPath.InConsultationProjects,
					apiKey
				)
			)?.embedded?.niceIndevInconsultationProduct || []
	);

/**
 * Gets a project detail.
 *
 */

export const getProjectDetail = async (
	inDevReference: string
): Promise<ProjectDetail | ErrorResponse> =>
	//TODO don't cache error response
	await getFeedBodyCached<ProjectDetail | ErrorResponse>(
		cacheKeyPrefix,
		FeedPath.ProjectDetail + inDevReference,
		longTTL,
		async () =>
			await getFeedBodyUnCached<ProjectDetail | ErrorResponse>(
				origin,
				FeedPath.ProjectDetail + inDevReference,
				apiKey
			)
	);
