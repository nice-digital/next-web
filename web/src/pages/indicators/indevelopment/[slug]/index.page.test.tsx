import { render, screen, waitFor, within } from "@testing-library/react";
import { type GetServerSidePropsContext } from "next";
import { useRouter } from "next/router";

import { FeedPath, TopicSelectionReason } from "@/feeds/inDev/types";
import gidng10237 from "@/mockData/inDev/project/gid-ng10237.json";
import gidta10992 from "@/mockData/inDev/project/gid-ta10992.json";
import { addDefaultJSONFeedMocks, axiosJSONMock } from "@/test-utils/feeds";

import InDevelopmentPage, {
	getServerSideProps,
	InDevelopmentPageProps,
} from "./index.page";

const productRoot = "indicators",
	slug = "gid-ng10237",
	resolvedUrl = `/${productRoot}/indevelopment/${slug}`,
	getServerSidePropsContext = {
		params: {
			slug,
		},
		resolvedUrl,
		query: { productRoot },
	} as unknown as GetServerSidePropsContext<{ slug: string }>;

describe("/indevelopment/[slug].page", () => {
	beforeEach(() => {
		(useRouter as jest.Mock).mockImplementation(() => ({
			asPath: resolvedUrl,
		}));
	});

	describe("IndevelopmentPage", () => {
		let props: InDevelopmentPageProps;
		beforeEach(async () => {
			props = (
				(await getServerSideProps(getServerSidePropsContext)) as {
					props: InDevelopmentPageProps;
				}
			).props;
		});

		it("should match snapshot for main content", () => {
			render(<InDevelopmentPage {...props} />);
			expect(document.body).toMatchSnapshot();
		});

		it("should render the page title with reversed breadcrumbs for SEO", async () => {
			render(<InDevelopmentPage {...props} />);

			await waitFor(() => {
				expect(document.title).toEqual(
					`Project information | Adrenal insufficiency: acute and long-term management | Indicators | Standards and Indicators`
				);
			});
		});

		it("should render a non-linked breadcrumb for the project id", () => {
			render(<InDevelopmentPage {...props} />);
			const navigation = screen.getByRole("navigation", {
				name: "Breadcrumbs",
			});
			const idBreadcrumb = within(navigation).getByText("GID-NG10237");
			expect(idBreadcrumb).toBeInTheDocument();

			expect(
				screen.queryByRole("link", { name: "GID-NG10237" })
			).not.toBeInTheDocument();
		});

		it("should render the project description", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-ipg10248",
					},
					resolvedUrl: "/indicators/indevelopment/gid-ipg10248",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);
			expect(
				screen.getByText(
					/description: the aim of intramuscular diaphragm stimulation is to make the diaphragm contract, strengthening it and allowing full or partial weaning from mechanical ventilation\. this procedure needs intact phrenic nerve function and avoids the need to access the phrenic nerve through the neck or thorax, as well as reducing the risk of phrenic nerve damage\. the procedure is done laparoscopically with the patient under general anaesthesia\. a probe is used to identify areas of the diaphragm where minimal electrical stimulation causes maximal diaphragm contraction \(known as the 'motor points'\)\. two intramuscular electrodes are implanted on the abdominal surface of each hemi-diaphragm at the motor points\. the electrode leads are tunnelled subcutaneously to an exit site in the chest where they are connected to an external battery-powered pulse generator\. a reference electrode \(anode\) is also implanted and the leads tunnelled with the other electrodes\. intraoperative stimulation and voltage calibration tests are done to confirm adequate contraction of the diaphragm\. after implantation the patient has a diaphragm conditioning programme, which involves progressive use of the system for increasing periods of time with gradual weaning from the ventilator\./i
				)
			).toBeInTheDocument();
		});

		it("should render the project id number", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-hst10009",
					},
					resolvedUrl: "/indicators/indevelopment/gid-hst10009",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);
			expect(screen.getByText("ID number: 927")).toBeInTheDocument();
		});

		it("should render a project information overview heading h1", () => {
			render(<InDevelopmentPage {...props} />);

			expect(
				screen.getByRole("heading", {
					level: 1,
					name: "Adrenal insufficiency: acute and long-term management",
				})
			).toBeInTheDocument();
		});

		it.each([
			["Monitor", TopicSelectionReason.Monitor],
			["Anticipate", TopicSelectionReason.Anticipate],
			["NotEligible", TopicSelectionReason.NotEligible],
			["FurtherDiscussion", TopicSelectionReason.FurtherDiscussion],
		])(
			"should render text when a topic selection reason %s exists, topic reason text: %s",
			async (topicSelectionReason, topicSelectionReasonText) => {
				axiosJSONMock.reset();
				axiosJSONMock.onGet(new RegExp(FeedPath.ProjectDetail)).reply(200, {
					...gidta10992,
					topicSelectionReason: topicSelectionReason,
				});
				addDefaultJSONFeedMocks();

				props = (
					(await getServerSideProps({
						...getServerSidePropsContext,
						params: {
							slug: "gid-ta10992",
						},
						resolvedUrl: "/indicators/indevelopment/gid-ta10992",
					})) as {
						props: InDevelopmentPageProps;
					}
				).props;

				render(<InDevelopmentPage {...props} />);

				expect(screen.getByTestId("reason")).toBeInTheDocument();

				expect(screen.getByTestId("reason")).toHaveTextContent(
					`Reason for decision: ${topicSelectionReasonText}`
				);
			}
		);

		it("should render 'Developed as: [Process]' when status != TopicSelection and ProjectType='NG' ", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-ng10163",
					},
					resolvedUrl: "/indicators/indevelopment/gid-ng10163",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			expect(screen.getByText("Developed as: APG")).toBeInTheDocument();
		});

		it("should render 'Process: [Process]' when status != TopicSelection and ProjectType != 'NG'", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-hst10044",
					},
					resolvedUrl: "/indicators/indevelopment/gid-hst10044",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);
			expect(screen.getByText("Process: HST")).toBeInTheDocument();
		});

		it("should reder 'Notification date' when Process =='MT' ", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-mt130",
					},
					resolvedUrl: "/indicators/indevelopment/gid-mt130",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);
			expect(screen.getByText("Process: MT")).toBeInTheDocument();
			expect(
				screen.getByText("Notification date: February 2010")
			).toBeInTheDocument();
		});

		it("should render 'Referral date' when Process != 'MT ", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-tag383",
					},
					resolvedUrl: "/indicators/indevelopment/gid-tag383",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			expect(screen.getByText("Referral date:").tagName).toBe("P");
			const dates = screen.getAllByText("1 November 2005")[0] as HTMLElement;
			expect(dates.tagName).toBe("TIME");
			//TODO Check datetime attribute format
			// expect(dates).toHaveAttribute("datetime", "2020-05-11");
		});

		it("should render related links", async () => {
			//TODO check formatting and location of related links
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "cg189-update-1",
					},
					resolvedUrl: "/indicators/indevelopment/cg189-update-1",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			expect(
				screen.getByText(
					"Weight Management: preventing, assessing and managing overweight and obesity (update)"
				)
			).toBeInTheDocument();

			expect(
				screen.getByText(
					"https://www.nice.org.uk/guidance/indevelopment/gid-ng10182/documents"
				)
			).toBeInTheDocument();
		});

		it("should render 'register an interest in this interventional procedure' link when projectType starts 'IPG' ", async () => {
			// gid-ip1046
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-ip1046",
					},
					resolvedUrl: "/indicators/indevelopment/gid-ip1046",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			const interventionLink = screen.getByRole("link", {
				name: "Register an interest in this interventional procedure",
			});

			expect(interventionLink).toBeInTheDocument();
			expect(interventionLink).toHaveAttribute(
				"href",
				"/about/what-we-do/our-Programmes/NICE-guidance/NICE-interventional-procedures-guidance/IP-register-an-interest"
			);
		});

		it("should render further information links", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-tag383",
					},
					resolvedUrl: "/indicators/indevelopment/gid-tag383",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			const furtherInfoLink = screen.getByRole("link", {
				name: "CHTE processes and methods manual default",
			});

			expect(
				screen.getByText(
					"For further information on our processes and methods, please see our"
				)
			).toBeInTheDocument();

			expect(furtherInfoLink).toBeInTheDocument();
			expect(furtherInfoLink).toHaveAttribute(
				"href",
				"https://www.nice.org.uk/process/pmg20/chapter/introduction"
			);
		});

		it("should render email enquiries mailto link", () => {
			render(<InDevelopmentPage {...props} />);

			const link = screen.getByRole("link", {
				name: "hypoadrenalism@nice.org.uk",
			});

			expect(link).toBeInTheDocument();

			expect(link).toHaveAttribute("href", "mailto:hypoadrenalism@nice.org.uk");
		});

		it("should render topic areas", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-tag360",
					},
					resolvedUrl: "/indicators/indevelopment/gid-tag360",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			expect(screen.getByText("Topic area:")).toBeInTheDocument();

			const topicAreaList = screen.getByRole("list", { name: "Topic areas" });
			expect(within(topicAreaList).getByText("Cancer")).toBeInTheDocument();
			expect(
				within(topicAreaList).getByText("Digestive system")
			).toBeInTheDocument();
		});

		it("should render full update information links", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-qs10170",
					},
					resolvedUrl: "/indicators/indevelopment/gid-qs10170",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);
			expect(
				screen.getByText("This guidance will fully update the following:")
			).toBeInTheDocument();
			const updateLink = screen.getByRole("link", {
				name: "Acute kidney injury (QS76)",
			});
			expect(updateLink).toBeInTheDocument();
			expect(updateLink).toHaveAttribute("href", "/guidance/qs76");
		});

		it("should render partial update information links", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-ng10364",
					},
					resolvedUrl: "/indicators/indevelopment/gid-ng10364",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);
			const updateLink = screen.getByRole("link", {
				name: "Atopic eczema in under 12s: diagnosis and management (CG57)",
			});
			expect(
				screen.getByText("This guidance will partially update the following:")
			).toBeInTheDocument();
			expect(updateLink).toBeInTheDocument();
			expect(updateLink).toHaveAttribute("href", "/guidance/cg57");
		});

		it("should render consultee stakeholder items", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-hst10037",
					},
					resolvedUrl: "/indicators/indevelopment/gid-hst10037",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			expect(
				screen.getByRole("heading", { level: 3, name: "Stakeholders" })
			).toBeInTheDocument();

			const consulteeList = screen.getByLabelText("Consultee stakeholders");
			const { getAllByRole } = within(consulteeList);
			const consulteeListItems = getAllByRole("definition");
			expect(consulteeListItems.length).toBe(44);
		});

		it("should render commentator stakeholder items", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-hst10037",
					},
					resolvedUrl: "/indicators/indevelopment/gid-hst10037",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			expect(
				screen.getByRole("heading", { level: 3, name: "Stakeholders" })
			).toBeInTheDocument();

			const commentatorList = screen.getByLabelText("Commentator stakeholders");
			const { getAllByRole } = within(commentatorList);
			const commentatorListItems = getAllByRole("definition");
			expect(commentatorListItems.length).toBe(30);
		});

		it("should render legacy stakeholder items", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-mt130",
					},
					resolvedUrl: "/indicators/indevelopment/gid-mt130",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			const legacyStakeholderList = screen.getAllByRole("list", {
				name: "Legacy stakeholders",
			});

			expect(legacyStakeholderList).toMatchSnapshot();

			expect(
				screen.getByRole("heading", { level: 3, name: "Stakeholders" })
			).toBeInTheDocument();
		});

		it("should render suspend discontinued reason", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-ip1153",
					},
					resolvedUrl: "/indicators/indevelopment/gid-ip1153",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);
			expect(screen.getByTestId("suspendDiscontinuedReason")).toHaveTextContent(
				"Discontinued"
			);
		});

		it("should render project team", () => {
			render(<InDevelopmentPage {...props} />);
			expect(
				screen.getByRole("heading", { level: 3, name: "Project Team" })
			).toBeInTheDocument();

			const projectTeamList = screen.getByLabelText("Project team");
			const { getAllByRole, getByText } = within(projectTeamList);
			const projectTeamItems = getAllByRole("definition");
			expect(projectTeamItems.length).toBe(1);
			expect(getByText("Developer").tagName).toBe("DT");
			expect(getByText("National Guideline Centre").tagName).toBe("DD");
		});

		it("should render provisional schedule", async () => {
			props = (
				(await getServerSideProps({
					...getServerSidePropsContext,
					params: {
						slug: "gid-dg10049",
					},
					resolvedUrl: "/indicators/indevelopment/gid-dg10049",
				})) as {
					props: InDevelopmentPageProps;
				}
			).props;
			render(<InDevelopmentPage {...props} />);

			expect(
				screen.getByRole("heading", { level: 3, name: "Provisional Schedule" })
			).toBeInTheDocument();

			const provisionalScheduleList = screen.getByLabelText(
				"Provisional schedule"
			);
			const { getAllByRole, getByText } = within(provisionalScheduleList);
			const provisionalScheduleItems = getAllByRole("definition");
			expect(provisionalScheduleItems.length).toBe(2);
			expect(getByText("Committee meeting: 2").tagName).toBe("DT");
			expect(getByText("18 January 2023").tagName).toBe("DD");
		});

		it.todo("should render timeline");

		describe("ProjectHeading", () => {
			it("should render expected publication date metadata as time tag with correct formatted date", () => {
				render(<InDevelopmentPage {...props} />);

				expect(screen.getByText("Expected publication date").tagName).toBe(
					"SPAN"
				);

				const dates = screen.getAllByText("11 April 2024")[0] as HTMLElement;

				expect(dates.tagName).toBe("TIME");
				//TODO Check datetime attribute format
				// expect(dates).toHaveAttribute("datetime", "2020-05-11");
			});

			it("should render pageheader meta expected publication date TBC", async () => {
				props = (
					(await getServerSideProps({
						...getServerSidePropsContext,
						params: {
							slug: "gid-ta11036",
						},
						resolvedUrl: "/indicators/indevelopment/gid-ta11036",
					})) as {
						props: InDevelopmentPageProps;
					}
				).props;
				render(<InDevelopmentPage {...props} />);
				expect(
					screen.getByText("Expected publication date: TBC")
				).toBeInTheDocument();
			});

			it("should not render pageheader meta expected publication date if status = discontinued", async () => {
				props = (
					(await getServerSideProps({
						...getServerSidePropsContext,
						params: {
							slug: "gid-mt130",
						},
						resolvedUrl: "/indicators/indevelopment/gid-mt130",
					})) as {
						props: InDevelopmentPageProps;
					}
				).props;
				render(<InDevelopmentPage {...props} />);

				expect(
					screen.queryByText("Expected publication date: TBC")
				).not.toBeInTheDocument();

				expect(
					screen.queryByText("GID-MT130 ", {
						selector: ".page-header__metadata li",
					})
				).not.toBeInTheDocument();
			});

			it("should render pageheader meta discontinued if status = discontinued", async () => {
				props = (
					(await getServerSideProps({
						...getServerSidePropsContext,
						params: {
							slug: "gid-mt130",
						},
						resolvedUrl: "/indicators/indevelopment/gid-mt130",
					})) as {
						props: InDevelopmentPageProps;
					}
				).props;
				render(<InDevelopmentPage {...props} />);

				expect(
					screen.getByText("Discontinued GID-MT130", {
						selector: ".page-header__metadata li",
					})
				).toBeInTheDocument();

				expect(screen.getByText("Status: Discontinued")).toBeInTheDocument();
			});

			it("should render a 'register as a stakeholder' link", async () => {
				render(<InDevelopmentPage {...props} />);
				const stakeholderLink = screen.getByRole("link", {
					name: "Register as a stakeholder",
				});
				expect(stakeholderLink).toBeInTheDocument();
				expect(stakeholderLink).toHaveAttribute(
					"href",
					"https://alpha.nice.org.uk/get-involved/stakeholder-registration/register"
				);
			});
		});
	});

	describe("getServerSideProps", () => {
		it("should return a correct props when supplied with a valid slug", async () => {
			await expect(
				getServerSideProps(getServerSidePropsContext)
			).resolves.toMatchSnapshot();
		});

		describe("Redirects", () => {
			it("should return permanent redirect object to the published product URL when project status is 'Complete'", async () => {
				axiosJSONMock.reset();
				axiosJSONMock
					.onGet(new RegExp(FeedPath.ProjectDetail))
					.reply(200, { ...gidng10237, status: "Complete" });
				addDefaultJSONFeedMocks();

				await expect(
					getServerSideProps(getServerSidePropsContext)
				).resolves.toStrictEqual({
					redirect: {
						destination: "/indicators/gid-ng10237-",
						permanent: true,
					},
				});
			});

			it("should return not found if project doesn't exist", async () => {
				const notFoundIdSlug = "abc123";

				axiosJSONMock.reset();
				axiosJSONMock.onGet(new RegExp(FeedPath.ProjectDetail)).reply(404, {
					Message: "Not found",
					StatusCode: "NotFound",
				});
				addDefaultJSONFeedMocks();

				expect(
					await getServerSideProps({
						...getServerSidePropsContext,
						params: { slug: notFoundIdSlug },
					})
				).toStrictEqual({
					notFound: true,
				});
			});
			it.todo("should reject when request is incorrect");
		});
	});
});
