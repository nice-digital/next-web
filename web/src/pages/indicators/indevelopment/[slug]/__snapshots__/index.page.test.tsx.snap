// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/indevelopment/[slug].page IndevelopmentPage should match snapshot for main content 1`] = `
<body>
  <div>
    <nav
      aria-label="Breadcrumbs"
      role="navigation"
    >
      <ol
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/standards-and-indicators"
          >
            Standards and Indicators
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/standards-and-indicators/indicators"
          >
            Indicators
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/standards-and-indicators/indicators/indevelopment"
          >
            In development
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <span>
            GID-NG10237
          </span>
        </li>
      </ol>
    </nav>
    <script
      type="application/ld+json"
    >
      {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/",
        "name": "Home"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/standards-and-indicators",
        "name": "Standards and Indicators"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/standards-and-indicators/indicators",
        "name": "Indicators"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "@id": "/standards-and-indicators/indicators/indevelopment",
        "name": "In development"
      }
    },
    {
      "@type": "ListItem",
      "position": 5,
      "item": {
        "name": "GID-NG10237"
      }
    }
  ]
}
    </script>
    <div
      class="page-header"
      id="content-start"
    >
      <h1
        class="page-header__heading page-header__heading--alt"
      >
        Adrenal insufficiency: acute and long-term management
      </h1>
      <ul
        class="page-header__metadata"
      >
        <li>
          In development GID-NG10237
        </li>
        <li>
          <span>
            Expected publication date
          </span>
           
          <time
            datetime="11 April 2024"
          >
            11 April 2024
          </time>
        </li>
        <li>
          <a
            href="https://alpha.nice.org.uk/get-involved/stakeholder-registration/register"
          >
            Register as a stakeholder
          </a>
        </li>
      </ul>
    </div>
    <button>
      Read the consultation documents
    </button>
    <p>
      Status: 
      InProgress
    </p>
    <p>
      Decision: 
      Selected
       
    </p>
    <p>
      Developed as: 
      CG
    </p>
    <h3>
      Provisional Schedule
    </h3>
    <dl
      aria-label="Provisional schedule"
    >
      <dt>
        Expected publication
      </dt>
      <dd>
        11 April 2024
      </dd>
    </dl>
    <h3>
      Project Team
    </h3>
    <dl
      aria-label="Project team"
    >
      <dt>
        Developer
      </dt>
      <dd>
        National Guideline Centre
      </dd>
    </dl>
    <h3>
      Email enquiries
    </h3>
    <span>
      If you have any queries please email
    </span>
    <ul>
      <li>
        <a
          href="mailto:hypoadrenalism@nice.org.uk"
        >
          hypoadrenalism@nice.org.uk
        </a>
      </li>
    </ul>
    <h3>
      Timeline
    </h3>
    <p>
      Key events during the development of the guidance:
    </p>
    <table
      class="table table-condensed"
    >
      <tbody>
        <tr>
          <th>
            Date
          </th>
          <th>
            Update
          </th>
        </tr>
        <tr>
          <td>
            13 September 2022
          </td>
          <td>
            Stakeholder list updated
          </td>
        </tr>
        <tr>
          <td>
            07 July 2022
          </td>
          <td>
            Committee list updated
          </td>
        </tr>
        <tr>
          <td>
            25 May 2022
          </td>
          <td>
            Committee meeting
          </td>
        </tr>
        <tr>
          <td>
            14 April 2022
          </td>
          <td>
            Scope published
          </td>
        </tr>
        <tr>
          <td>
            01 February 2022 - 01 March 2022
          </td>
          <td>
            Draft scope consultation
          </td>
        </tr>
        <tr>
          <td>
            01 February 2022 - 01 March 2022
          </td>
          <td>
            Committee member recruitment
          </td>
        </tr>
        <tr>
          <td>
            01 February 2022
          </td>
          <td>
            Guideline title changed from Hypoadrenalism: acute and long-term management to Adrenal insufficiency: acute and long-term management.
          </td>
        </tr>
        <tr>
          <td>
            05 January 2022
          </td>
          <td>
            Scoping workshop
          </td>
        </tr>
        <tr>
          <td>
            18 October 2021 - 25 October 2021
          </td>
          <td>
            Topic expert committee member recruitment
          </td>
        </tr>
      </tbody>
    </table>
    <p>
      For further information on the guideline development process, please read
       
      <a
        href="https://www.nice.org.uk/about/what-we-do/our-programmes/nice-guidance/nice-guidelines/how-we-develop-nice-guidelines"
      >
        how we develop NICE guidelines
      </a>
    </p>
  </div>
</body>
`;

exports[`/indevelopment/[slug].page IndevelopmentPage should render legacy stakeholder items 1`] = `
Array [
  <ul
    aria-label="Legacy stakeholders"
  >
    <li>
      &lt;p&gt;
                    	
                    		
                        	
                        		&lt;p&gt;NICE is developing guidance on this technology. If you would like to become a registered stakeholder in the development of guidance on this technology, please complete a &lt;a href="http://www.nice.org.uk/media/default/about/what-we-do/nice-guidance/nice-medical-technologies/mtep-notification-form.docx" nice-original-href="http://www4.nice.org.uk/Media/Default/About/what-we-do/NICE-guidance/NICE-medical-technologies/MTEP-notification-form.docx"&gt;stakeholder registration and confidentiality form&lt;/a&gt; (Word), and return to us at &lt;a href="mailto:medtech@nice.org.uk"&gt;medtech@nice.org.uk&lt;/a&gt;.&lt;/p&gt;
&lt;p&gt;Registered stakeholders will be informed of key stages in the development of guidance including when documents go out to consultation or comment.&lt;/p&gt;

                        	
                        	
                        
                    &lt;/p&gt;
    </li>
  </ul>,
]
`;

exports[`/indevelopment/[slug].page getServerSideProps should return a correct props when supplied with a valid slug 1`] = `
Object {
  "props": Object {
    "description": null,
    "developedAs": "CG",
    "evidenceAssessmentGroup": null,
    "fullUpdates": Array [],
    "idNumber": null,
    "indevCommentators": Array [],
    "indevConsultees": Array [],
    "indevEmailEnquiries": Array [
      Object {
        "eTag": null,
        "item": "hypoadrenalism@nice.org.uk",
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
    ],
    "indevLegacyStakeholders": Array [],
    "indevProcessHomepage": Object {
      "description": "For further information on the guideline development process, please read",
      "eTag": null,
      "linkText": "how we develop NICE guidelines",
      "links": Object {
        "self": Array [
          Object {
            "href": "https://www.nice.org.uk/about/what-we-do/our-programmes/nice-guidance/nice-guidelines/how-we-develop-nice-guidelines",
          },
          Object {},
        ],
      },
    },
    "indevProjectRelatedLinks": Array [],
    "indevProjectTeamMembers": Array [
      Object {
        "column1": "Developer",
        "column2": "National Guideline Centre",
        "eTag": null,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
    ],
    "indevScheduleItems": Array [
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": null,
        "column1": "Expected publication",
        "column2": "11 April 2024",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
    ],
    "indevStakeholderRegistration": Array [
      Object {
        "href": "https://alpha.nice.org.uk/get-involved/stakeholder-registration/register",
      },
    ],
    "indevTimelineItems": Array [
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": null,
        "column1": "13 September 2022",
        "column2": "Stakeholder list updated",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": null,
        "column1": "07 July 2022",
        "column2": "Committee list updated",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": "Meeting number",
        "column1": "25 May 2022",
        "column2": "Committee meeting",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": null,
        "column1": "14 April 2022",
        "column2": "Scope published",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": null,
        "column1": "01 February 2022 - 01 March 2022",
        "column2": "Draft scope consultation",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": null,
        "column1": "01 February 2022 - 01 March 2022",
        "column2": "Committee member recruitment",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": null,
        "column1": "01 February 2022",
        "column2": "Guideline title changed from Hypoadrenalism: acute and long-term management to Adrenal insufficiency: acute and long-term management.",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": "Location of meeting",
        "column1": "05 January 2022",
        "column2": "Scoping workshop",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
      Object {
        "additionalInfo": null,
        "additionalInfoLabel": null,
        "column1": "18 October 2021 - 25 October 2021",
        "column2": "Topic expert committee member recruitment",
        "eTag": null,
        "hidden": false,
        "links": Object {
          "self": Array [
            Object {},
          ],
        },
      },
    ],
    "indevTopicItems": Array [],
    "partialUpdates": Array [],
    "process": "CG",
    "projectType": "NG",
    "reference": "GID-NG10237",
    "referralDate": null,
    "status": "InProgress",
    "summary": null,
    "suspendDiscontinuedReason": null,
    "suspendDiscontinuedUrl": null,
    "suspendDiscontinuedUrlText": null,
    "technologyType": null,
    "title": "Adrenal insufficiency: acute and long-term management",
    "topicSelectionDecision": "Selected",
    "topicSelectionFurtherInfo": null,
    "topicSelectionReasonText": null,
  },
}
`;
